spark-3.1-spanner-lib/src/main/java/com/google/cloud/spark/spanner/SpannerScanner.java

```diff
@@ -94,11 +94,12 @@ public InputPartition[] planInputPartitions() {
       sqlStmt += " WHERE " + SparkFilterUtils.getCompiledFilter(true, Optional.empty(), filters);
     }
 
-    Boolean enableDataboost = this.opts.get("enableDataboost") == "true";
+    Boolean enableDataboost = this.opts.get("enableDataBoost").equalsIgnoreCase("true");
```

Comment:
What happens if null is returned? Won't this crash? I think we shall need to either use StringUtils.equalsIgnoreCase https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#equalsIgnoreCase-java.lang.CharSequence-java.lang.CharSequence- or have to check that the value is non-null before the equalsCaseFold

Reply:
Thanks, Emmanuel. I am modifying the pull request to fix it.